API-GUIDELINE.md

```diff
@@ -0,0 +1,254 @@
+             ],
+             "images": [
+               {
+                 "url": "https://www.commercetools.com/cli/data/252542005_1.jpg",
+                 "dimensions": {
+                   "w": 1400,
+                   "h": 1400
+                 }
+               }
+             ],
+             "attributes": [],
+             "assets": []
+           },
+           "price": {
+             "value": {
+               "type": "centPrecision",
+               "fractionDigits": 2,
+               "currencyCode": "EUR",
+               "centAmount": 2800
+             },
+             "id": "8da659ef-9e54-447d-9c36-84912db1848f"
+           },
+           "quantity": 2,
+           "discountedPricePerQuantity": [],
+           "state": [
+             {
+               "quantity": 2,
+               "state": {
+                 "typeId": "state",
+                 "id": "7c2e2694-aefe-43d7-888e-6a99514caaca"
+               }
+             }
+           ],
+           "priceMode": "Platform",
+           "lineItemMode": "Standard",
+           "totalPrice": {
+             "type": "centPrecision",
+             "fractionDigits": 2,
+             "currencyCode": "EUR",
+             "centAmount": 5600
+           }
+         }
+       ],
+       "store": {
+         "typeId": "store",
+         "key": "test-key"
+       },
+       "cartState": "Active",
+       "totalPrice": {
+         "type": "centPrecision",
+         "fractionDigits": 2,
+         "currencyCode": "EUR",
+         "centAmount": 5600
+       },
+       "customLineItems": [],
+       "discountCodes": [],
+       "inventoryMode": "None",
+       "taxMode": "Platform",
+       "taxRoundingMode": "HalfEven",
+       "taxCalculationMode": "LineItemLevel",
+       "refusedGifts": [],
+       "origin": "Customer"
+     }
+
+##Naming convention
+
+###General Guidelines
+To use consistent names these are the rules to follow:
+
+ DO'S
+ - use nouns instead of verbs to expose resources
+ - use lowercase and separated by hyphens (-) for path segments like /shopping-lists
+ - use American English
+ - create each resource as unique resource name
+ - use the proper HTTP methods for API operations whenever possible
+ 
+ DON'TS
+ - DON'T use snake case (_) or other punctuation
+ - DON'T use file extensions
+ - DON'T use abbreviations
+ - DON'T use CRUD HTTP methods or verbs
+
+###Naming Properties
+The properties are identified in base of the type: 
+ - DateTime
+ - Boolean
+ - Number/Integer
+ - String
+ - Array
+ - Object
+
+####DateTime
+In general, define a DateTime property using a name ending with "At", then if there is a range of date use "From" and "To".
+
+Some examples:
+
+        returnAt
+        expirationAt
+        validFrom
+        validTo
+
+####Boolean
+Define a Boolean property using a name which includes "is" or "has" and the verb at past participle.
+
+Some examples:
+
+        isEmailVerified
+        hasStagedChanges
+        isPublished
+        isActive
```

Comment:
@jenschude you mentioned earlier sometime that we can recommend to write it without is or has as it might be redundant when we generate the SDKs.

e.g. active instead of isActive so that we generate a method named isActive based on the name.

WDYT ?

Reply:
It depends. On the pure API layer it's more speaking as the type can be read from the name.

For our SDKS its different. E.g. java says that a getter for a boolean bean should start with `is`. So we would have to declare mappings to avoid something like `isIsActive` or `getIsActive` (atm this is how it looks like in PHP)

And defacto the whole API already uses these prefixes and can't be easily changed

Reply:
so are you suggesting that I should change everything without "is"? Assuming that our guidelines is also for the future

Reply:
and what about something like "emailIsVerified"?

Reply:
if so it would be `emailVerified` -> singular rule ;)

And if we don't want to codify the state as is as best as possible, but make decisions about how it should be in the future, it should not only be us to do so ;)

Reply:
I agree but we should suggest something to the backend team, so I would go with `emailVerified`

Reply:
No we can't change existing fields. We could only propose how it should be done in the future.

Reply:
yes, this is what I was saying ;)